src/schema.ts

```diff
@@ -235,9 +177,69 @@ export interface MachineState {
 }
 
 /**
- * All possible events relating to the machine itself.
+ * Represents the state pertaining to the vibraphone.
  */
-export type MachineEvent = MachineMuteEvent;
+export interface VibraphoneState {
+	/** Whether or not the vibrato is engaged */
+	vibratoEnabled: boolean;
+	/** The vibrato speed from 0 to 1 */
+	vibratoSpeed: number;
+	/**
+	 * Which notes represent which channels.
+	 * These cannot be changed via events and thus stay
+	 * the same throughout the course of an entire [[Performance]].
+	 * They can however, be changed while editing a [[Program]].
+	 */
+	notes: { [VC in VibraphoneChannel]: Note };
+}
+
+/** Represents the state pertaining to the bass. */
+export interface BassState {
+	/**
+	 * Which frets the capos are on and if they are engaged.
+	 * `0` is an open string.
+	 */
+	capos: { [S in BassString]: { engaged: boolean, fret: number } };
+	/**
+	 * Which notes the strings are tuned to.
+	 * Nothing means the standard bass tuning:
+	 * ```typescript
+	 * {
+	 *   4: "E1",
+	 *   3: "A1",
+	 *   2: "D2",
+	 *   1: "G2",
+	 * }
+	 * ```
+	 */
+	tuning: { [S in BassString]?: Note };
+}
+
+/** Represents the state pertaining to the hihat machine. */
+export interface HihatMachineState {
+	/** The meaning of this property has yet to be determined. */
+	setting: string;
+}
+
+/** Represents the state pertaining to the hihat, not the hihat machine. */
+export interface HihatState {
+	/** Whether or not the hihat is closed. */
+	closed: boolean;
+}
+
+/** Represents the machine's state. Used to specify both the start state and running state of the machine. */
+export interface State {
+	machine: MachineState;
+	vibraphone: VibraphoneState;
+	bass: BassState;
+	hihatMachine: HihatMachineState;
+	hihat: HihatState;
+}
+
+//#endregion
+
+//#region Events
+
 /** An event representing the muting or unmuting of a [[Channel]] */
 export interface MachineMuteEvent {
```

Comment:
Should be changed to "ChannelMuteEvent" if it is the muting/unmuting of a Channel.

Reply:
Sure